x-pack/plugins/ingest_manager/server/services/output.ts

```diff
@@ -16,11 +17,17 @@ class OutputService {
       type: OUTPUT_SAVED_OBJECT_TYPE,
       filter: `${OUTPUT_SAVED_OBJECT_TYPE}.attributes.is_default:true`,
     });
+    const cloud = appContextService.getCloud();
+    const cloudId = cloud?.isCloudEnabled && cloud.cloudId;
+    const cloudUrl = cloudId && decodeCloudId(cloudId)?.elasticsearchUrl;
+    const flagsUrl = appContextService.getConfig()!.fleet.elasticsearch.host;
+    const defaultUrl = 'http://localhost:9200';
+    const defaultOutputUrl = cloudUrl || flagsUrl || defaultUrl;
```

Comment:
shouldn't it be `flagsUrl || cloudUrl || defaultUrl`? (prefer flagsUrl over everything else including cloudUrl)

Reply:
I think if we are in the context of cloud `cloudUrl` is probably the better bet also you should not be able to edit the flags as these flags are not whitelisted

Reply:
ok, I wasn't clear on where we were with having our settings whitelisted in cloud. if they are not going to end up whitelisted then my comment isn't relevant 😄

Reply:
@jen-huang The logic was initially discussed in Slack & https://github.com/elastic/kibana/issues/64513#issuecomment-622087849

@nchaulet is correct that we cannot use these flags in Cloud. We should only have a cloudId if we're in cloud and, at least for now, if we're in Cloud we do not want users pointing to other instances.